src/core/ActivityInfoTree.cpp

```diff
@@ -283,5 +302,70 @@ void ActivityInfoTree::init()
 
     qmlRegisterSingletonType<QObject>("GCompris", 1, 0, "ActivityInfoTree", menuTreeProvider);
     qmlRegisterType<ActivityInfo>("GCompris", 1, 0, "ActivityInfo");
+
+
+}
+void ActivityInfoTree::filterBySearch(const QString& text,int cursorPosition)
```

Comment:
the cursor position shouldn't be a condition on the filtering. There is no "logical" reason to have it here. (I understand you use it for the if-case but it does not fit here)